### DISCUSSION TITLE
Clear cache of dependency lookups

### DISCUSSION THREAD

**Discussion by ryancausey** [renovatebot/renovate#10690] [Request Help] [answered]
I'm currently working through an issue raised by the bot in the onboarding MR:

⚠ Dependency Lookup Warnings ⚠

Please correct - or verify that you can safely ignore - these lookup failures before you merge this MR.

Failed to look up dependency public.ecr.aws/lambda/python

Files affected: lambda/Dockerfile, lambda/test/Dockerfile

I dug into the logs and found that the AWS public ECR look ups don't work, and there's an active issue to support AWS public ECR (#9268). I have since switched the dockerfiles to instead use the dockerhub mirror for the AWS lambda base images, which I believe should be supported.

However, on subsequently running the pipeline, it seems to remember that the lookup was against AWS ECR public and continues to complain in the logs. How do I get the renovate job to forget the lookup and try again using the changed dockerfiles?

This is all running in GitLab SAAS, using the renovate-runner recommended project setup.

**Comment by rarkins** [comment, marked as answer]
I don't think this is likely to be a cache issue and even less likely if you're using a self hosted Gitlab runner. Run with debug logging and check the "packageFiles with updates" message in particular.

**Reply by ryancausey** [reply]
So the relevant log snippet:

<details>
  <summary>Log snippet</summary>

  ```
DEBUG: packageFiles with updates (repository=stripe-webhook-receiver)
       "config": {
         "dockerfile": [
           {
             "packageFile": "lambda/Dockerfile",
             "deps": [
               {
                 "depName": "amazon/aws-lambda-python",
                 "currentValue": "3.8",
                 "replaceString": "amazon/aws-lambda-python:3.8",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "stage",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "versioning": "docker",
                 "currentVersion": "3.8",
                 "fixedVersion": "3.8"
               },
               {
                 "depName": "public.ecr.aws/lambda/python",
                 "currentValue": "3.8",
                 "replaceString": "public.ecr.aws/lambda/python:3.8",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "depIndex": 1,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "public.ecr.aws/lambda/python",
                     "message": "Failed to look up dependency public.ecr.aws/lambda/python"
                   }
                 ],
                 "versioning": "docker"
               }
             ]
           },
           {
             "packageFile": "lambda/test/Dockerfile",
             "deps": [
               {
                 "depName": "amazon/aws-lambda-python",
                 "currentValue": "3.8",
                 "replaceString": "amazon/aws-lambda-python:3.8",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "stage",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "versioning": "docker",
                 "currentVersion": "3.8",
                 "fixedVersion": "3.8"
               },
               {
                 "depName": "public.ecr.aws/lambda/python",
                 "currentValue": "3.8",
                 "replaceString": "public.ecr.aws/lambda/python:3.8",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "depIndex": 1,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "public.ecr.aws/lambda/python",
                     "message": "Failed to look up dependency public.ecr.aws/lambda/python"
                   }
                 ],
                 "versioning": "docker"
               }
             ]
           }
         ],
  ```
</details>

Why is the second entry in the `deps` array present still? I merged the dockerfile change into master and rebased the renovate onboarding MR on top of it. Can I safely ignore this warning?

**Reply by rarkins** [reply]
I suggest you restart the onboarding and this time don't rebase the onboarding branch in case that's what's confused it. To restart: rename the PR, close it and delete the branch before running Renovate again.

**Reply by ryancausey** [reply]
Same result. New MR has the same errors reported AWS ECR public repository is still in the list.

<details>
  <summary>Log snippet</summary>
  
  ```
DEBUG: packageFiles with updates (repository=stripe-webhook-receiver)
       "config": {
         "dockerfile": [
           {
             "packageFile": "lambda/Dockerfile",
             "deps": [
               {
                 "depName": "amazon/aws-lambda-python",
                 "currentValue": "3.8",
                 "replaceString": "amazon/aws-lambda-python:3.8",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "stage",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "versioning": "docker",
                 "currentVersion": "3.8",
                 "fixedVersion": "3.8"
               },
               {
                 "depName": "public.ecr.aws/lambda/python",
                 "currentValue": "3.8",
                 "replaceString": "public.ecr.aws/lambda/python:3.8",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "depIndex": 1,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "public.ecr.aws/lambda/python",
                     "message": "Failed to look up dependency public.ecr.aws/lambda/python"
                   }
                 ],
                 "versioning": "docker"
               }
             ]
           }
         ],
  ```
</details>

Dockerfile uses this from line:

`FROM amazon/aws-lambda-python:3.8 AS requirements_stage`

where before it was

`FROM public.ecr.aws/lambda/python:3.8 AS requirements_stage`

**Reply by rarkins** [reply]
I think the most likely explanation is that something in your repo is not as you think. If you check the renovate/configure branch manually, is it one commit ahead of your default branch?

**Reply by ryancausey** [reply]
It sure is:

```
commit 1708361845cdd674915b3aabd30e560938c55eae (HEAD -> renovate/configure, origin/renovate/configure)
Author: Renovate Bot <bot@renovateapp.com>
Date:   Fri Jul 2 14:55:55 2021 +0000

    Add renovate.json

commit 61ca45adbaadd71a4355e76b611a1d15910689bd (origin/master, origin/HEAD, master)
Author: Ryan Causey
Date:   Thu Jul 1 23:20:10 2021 -0700

    Use dockerhub paths since renovate can't pull from aws.ecr.public
```

I also confirmed the dockerfile change itself exists in the rennovate/configure branch.

**Reply by ryancausey** [reply]
Would it be useful to attach the entire tranche of debug logs for this project?

**Reply by rarkins** [reply]
Can you reproduce it with a new repo with a single dependency in a single Dockerfile?

**Reply by ryancausey** [reply]
I tried on this example repo and the MR created did not complain with the dockerfile having the `FROM amazon/aws-lambda-python:3.8`

https://gitlab.com/rccausey/renovate-aws-ecr-issue/-/merge_requests/1

Might be me being naive but it really seems like the renovate bot is somehow "remembering" there was a reference to public.ecr.aws at some point on the other repo. I tried clearing the runner cache and that didn't seem to work. I'm using gitlab shared runners not any of my own.

It doesn't look back at the recent commit history or anything does it?

**Reply by rarkins** [reply]
Are you sure you don't have a second reference to the same ecr image left in the repo?

**Reply by ryancausey** [reply]
![image](https://user-images.githubusercontent.com/2704075/124309665-efe28100-db1f-11eb-9489-4915b078726b.png)

This is why I need 4 sets of eyeballs or someone else to check me when I'm dumb. I missed the second stage build part and it still had the old, incorrect reference.
